Vagrantfile

```diff
@@ -1,19 +1,37 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
-require './lib/drupalvm/vagrant'
+require 'json'
 
-# Absolute paths on the host machine.
-host_drupalvm_dir = File.dirname(File.expand_path(__FILE__))
-host_project_dir = ENV['DRUPALVM_PROJECT_ROOT'] || host_drupalvm_dir
-host_config_dir = ENV['DRUPALVM_CONFIG_DIR'] ? "#{host_project_dir}/#{ENV['DRUPALVM_CONFIG_DIR']}" : host_project_dir
+drupalvm_env = ENV['DRUPALVM_ENV'] || 'vagrant'
 
-# Absolute paths on the guest machine.
-guest_project_dir = '/vagrant'
-guest_drupalvm_dir = ENV['DRUPALVM_DIR'] ? "/vagrant/#{ENV['DRUPALVM_DIR']}" : guest_project_dir
-guest_config_dir = ENV['DRUPALVM_CONFIG_DIR'] ? "/vagrant/#{ENV['DRUPALVM_CONFIG_DIR']}" : guest_project_dir
+# Default paths when the project is based on Drupal VM.
+host_project_dir = host_drupalvm_dir = host_config_dir = __dir__
+guest_project_dir = guest_drupalvm_dir = guest_config_dir = '/vagrant'
 
-drupalvm_env = ENV['DRUPALVM_ENV'] || 'vagrant'
+if File.exist?("#{host_project_dir}/composer.json")
+  cconfig = {}
+  composer_conf = JSON.parse(File.read("#{host_project_dir}/composer.json"))
+  if composer_conf['extra'] && composer_conf['extra']['drupalvm']
+    cconfig = composer_conf['extra']['drupalvm']
+  end
+
+  # If Drupal VM is a Composer dependency set the correct path.
+  vendor_dir = composer_conf.fetch('extra', {}).fetch('vendor-dir', 'vendor')
+  drupalvm_path = "#{vendor_dir}/geerlingguy/drupal-vm"
```

Comment:
>  There are some Windows developers who might prefer to code entirely within the VM right?

Yes. (And the occasional Mac or Linux user who likes the isolation.)

> Also this setup requires the user to run `composer install` to download Drupal VM. That's probably alright, I mean all window devs using this setup would have composer installed, right?

Preferably, yes. But right now there are a lot of Windows devs who neither want nor are allowed to set up PHP and/or Composer on Windows... but it could be possible we could make that a requirement for 5.0.0+...